ispcrt/ispcrt.hpp

```diff
@@ -168,6 +169,10 @@ inline void* Device::nativePlatformHandle() const { return ispcrtPlatformNativeH
 inline void* Device::nativeDeviceHandle() const { return ispcrtDeviceNativeHandle(handle()); }
 inline void* Device::nativeContextHandle() const { return ispcrtDeviceContextNativeHandle(handle()); }
 
+inline ISPCRTDeviceType Device::getDeviceType() const {
```

Comment:
In C API current name looks good (ispcrtGetDeviceType) but in C++ API you could also skip Device word in the middle:

Device::getType() looks more natural since it's method executed on Device.

Reply:
Sounds good. Same as before, shall we go with `Impl` to disambiguate with the internal interface in `ispcrt/detail/`?

Reply:
I don't understand @dbabokin why we can't have getType() method (same name) in these 2 classes:
- ispcrt::base::Device (detail -> implementation)
- ispcrt::Device (API / wrapper)

From my perspective this approach is valid (separation is done using namespace and different classes, method name can be same in both classes).

But if we can't have same name for these 2 classes then I would keep "clean version" in API and add change getType() -> getTypeImpl() for ispcrt::base::Device (which looks awkward to me anyway) but @dbabokin - it's your call.

Reply:
Sorry, I missed that they are class members. My bad! I was mislead by the name `getDeviceType()`, assuming that they belong to the `ispcrt` namespace, but not to the class. Then `getType()` in both cases are absolutely valid name and I think we should go with it.